### ISSUE TITLE
[roadmap] 2025 Q3: Optimizing User Experience

### ISSUE THREAD

**Issue by zhuzilin** [THUDM/slime#32]
Welcome to our first public roadmap! slime was previously developed and used as an in-house solution. Now that it is open-sourced, our primary goal is to make it accessible and useful for a wider community. To achieve this, our immediate priority is to significantly enhance the user experience. This roadmap outlines the key areas we are currently focusing on to make the framework more intuitive and powerful for everyone.

### SGLang Compatibility
Ensure slime runs seamlessly with native SGLang.
- [x] Merge core features into SGLang #6

### Make Megatron Faster and Easier to Use
Boost Megatron's training performance, with a focus on Mixture-of-Experts (MoE).
- [x] Implement correct Megatron offloading to resolve memory issues.
  - [x] Transfer to [torch_memory_saver](https://github.com/fzyzcjy/torch_memory_saver)
  - [x] Put all training within torch_memory_saver.
- [ ] MTP support #83 
  - [x] MTP rollout only.
- [ ] ~Add 1F1B-style forward-backward overlap.~ Megatron not ready.
- [ ] ~Enable FP8 training.~ Megatron not ready.
  - [ ] ~cuda 12.9 image.~ Use sglang cuda 12.9 image

User friendly
- [ ] https://github.com/THUDM/slime/issues/50
- [ ] Custom optimizer, e.g. muon, deepspeed cpu adam.

### Expanded Post-Training Capabilities
Evolve slime into a comprehensive post-training framework beyond GRPO RL.
- [x] Add Supervised Fine-Tuning (SFT) support.
  - [x] Add SFT example #58 
  - [ ] Support loading large datasets (Hugging Face Datasets, Parquet).
  - [ ] ~Enable evaluation only.~ USe Comment out the training part in `train.py` would be fine.
- [x] ~Add support for pairwise loss functions (e.g., Bradley-Terry, DPO).~ Added custom loss interface for now. #23
- [x] Support additional RL algorithms. #16

### Reliability
- [x] Establish CI/CD pipelines for robust testing and deployment.

### More User-Friendly Examples
- [x] Add a multi-turn conversation example (e.g., ~[RAGEN](https://github.com/RAGEN-AI/RAGEN.git)~ Search-R1) 
- [ ] Provide an example of an external agent environment. @yitianlian @jhinpan

### Interesting Reproductions
- [x] Search-R1 #53 
- [ ] ~RAGEN #52 @Qihang-Zhang~
- [ ]  ~[MemAgent](https://arxiv.org/abs/2507.02259)~

### ~PyTorch Native Backend~
- [ ] ~Develop an FSDP2-based backend to support loading and training any Hugging Face model.~ #74 
    Doesn't have enough resource to support this at the moment.

**Comment by zlH518** [comment]
Does this framework intend to expand the PPO algorithm? Because when I used the slim framework to implement PPO, I found that it seemed that it was impossible to make good incremental changes and sometimes even required major changes.

**Comment by zhuzilin** [comment]
It'll be hard to directly implement ppo cleanly.

I think there are several prerequisites we need to settle. for example,
1. how to pass arguments to multiple megatron models.
2. how to correctly implement the critic head in megatron without any intrusion to megatron.
3. if we hope to do colocate with ppo, then some sglang servers will need to be updates with ipc (from tensors), some with nccl (from distributed)

And I'm also not sure that as the model gets larger and larger, is it still necessary to implement ppo, for it will double the GPU resources. maybe we can settle with reinforce or reinforce++ for a moment.

**Comment by zcuuu** [comment]
hello,现在我看文档里面的介绍是 训推共卡。请问后续会探索  训推分离吗

**Comment by casper-hansen** [comment]
Linking #200 here as potentially small UX issue.

**Comment by zhuzilin** [comment]
@zcuuu 文档中也有训推分离

**Comment by zhuzilin** [comment]
close as v0.1.0 is released.

**Comment by CarnegieBin** [comment]
支持muon吗？
